services/venice-controller/src/main/java/com/linkedin/venice/controller/server/ControllerRoutes.java

```diff
@@ -105,6 +107,29 @@ public Route updateKafkaTopicLogCompaction(Admin admin) {
     });
   }
 
+  /**
+   * No ACL check; any user is allowed to check topic configs.
+   */
+  public Route getKafkaTopicConfigs(Admin admin) {
+    return (request, response) -> {
+      PubSubTopicConfigResponse responseObject = new PubSubTopicConfigResponse();
+      response.type(HttpConstants.JSON);
+
+      try {
+        responseObject.setCluster("");
+        PubSubTopic topicName = pubSubTopicRepository.getTopic(request.queryParams(TOPIC));
+        TopicManager topicManager = admin.getTopicManager();
+        PubSubTopicConfiguration pubSubTopicConfiguration = topicManager.getCachedTopicConfig(topicName);
```

Comment:
As the cached topic config inside the topic manager has 5 mins expire after write, so do we need to add some message inside the response to say that the topic config may not reflect the very recent change.

Reply:
Oh, good catch, let me use the other API instead; we should report the latest value in the tooling. The new command is supposed to serve for validation purpose, so it will be confusing if the query return a result different from the update command.